### ISSUE TITLE
Maliar loop with arbitrary loss function (FOC for automated test)

### ISSUE THREAD

**Issue by sbenthall** [scikit-agent/scikit-agent#105]
In principle, we should be able to run the Maliar training loop with any arbitrary loss function. (See #93 for changes in the loss function architecture and API).

For this issue:
 - create a loss function generator (ideally along the lines of #93)
 - create a test case based on an analytically solved case for which the FOC is known and sufficient for solving the model
 - use the Maliar loop with the hand-crafted FOC loss to solve the test case.
